de.persosim.driver.connector.ui/src/de/persosim/driver/connector/ui/parts/ReaderPart.java

```diff
@@ -320,9 +457,21 @@ public void run() {
 				});
         	}
         	break;
+        case KEYS_ALL_PINSAVER:
+        	for(final Button button : keysPinSaver) {
```

Comment:
This is the third copy of identical loops, thats definitely too much copied code